redis/src/commands/cluster_scan.rs

```diff
@@ -146,6 +146,9 @@ pub(crate) trait ClusterInScan {
 
     /// Check if all slots are covered by the cluster
     async fn are_all_slots_covered(&self) -> bool;
+
+    /// Check if the topology of the cluster has changed and refresh the slots if needed
+    async fn refresh_if_topology_changed(&self) -> RedisResult<()>;
```

Comment:
so shouldn't we remove the refresh_slots function from the trait?

Reply:
In retry scan (scan failed for connection reasons (no connection to address, master down and similar - we perform error handling and retry for specific errors)) we do refresh slots since we know something went wrong with the topology view we have.

Reply:
I mean, instead of duplication the logic of the periodic checks, you can move it to a separate function refresh_if_topology_changed and call it both for scan purposes and for the periodic checks, Then you can remove the refresh_slots requirement from the trait

Reply:
then you don't need to change check_for_topology_diff to pub(crate)